sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy

```diff
@@ -1835,12 +1835,31 @@ class BlobAPITest extends APISpec {
         bc.upload(defaultInputStream.get(), defaultDataSize)
 
         when:
-        bc.setTierWithResponse(AccessTier.HOT, new LeaseAccessConditions().leaseId("garbage"), null, null)
+        bc.setTierWithResponse(AccessTier.HOT, null, new LeaseAccessConditions().leaseId("garbage"), null, null)
 
         then:
         thrown(StorageException)
     }
 
+    @Unroll
+    def "Copy with tier"() {
+        setup:
+        def bc = cc.getBlockBlobClient(generateBlobName())
+        bc.uploadWithResponse(defaultInputStream.get(), defaultDataSize, null, null, tier1, null, null, null)
+        def bcCopy = cc.getBlockBlobClient(generateBlobName())
+
+        when:
+        bcCopy.copyFromURLWithResponse(new URL(bc.getBlobUrl().toString() + "?" + bc.generateSAS(OffsetDateTime.now().plusHours(1), new BlobSASPermission().read(true))), null, tier2, null, null, null, null)
+
+        then:
+        bcCopy.getPropertiesWithResponse(null, null, null).headers().value("x-ms-access-tier") == tier2.toString()
```

Comment:
>.headers().value("x-ms-access-tier")  [](start = 58, length = 37)

I think this is already stored in BlobProperties.accessTier

Reply:
It is not.

Reply:
Fixes this in #5268, will clean this up when I merge these changes into that branch